.github/workflows/ci.yml

```
@@ -244,6 +244,26 @@ jobs:
     - name: run Miri tests
       run: just miri cordyceps
 
+  # run no-atomics target tests
+  cordyceps_no_atomics_compat:
+    needs: changed_paths
+    if: needs.changed_paths.outputs.should_skip != 'true' || !fromJSON(needs.changed_paths.outputs.paths_result).cordyceps.should_skip
+    runs-on: ubuntu-latest
+    name: no-atomics compat (cordyceps)
+    steps:
+    - name: install rust toolchain
+      run: rustup show
+    - uses: actions/checkout@v4
+    - name: run check
+      # we add the target HERE instead of in the 'install rust toolchain' step
+      # because we need the `rust-toolchain.toml` file to be checked out, otherwise
+      # the target is only added for `stable`. We could also add this to the
+      # `targets` item in the toolchain file, but that would be a waste for all
+      # other tests.
+      run: |
+        rustup target add thumbv6m-none-eabi
+        cargo check -p cordyceps --target thumbv6m-none-eabi
```

a couple nits: i would prefer to do the `rustup target add` and `cargo check` parts in separate steps, so that it's more obvious which one failed in the github UI. also, rather than leaving the comment discussing how we need to do the target add step separately from the "rustup show" step due to needing the target file, why not just do this:
```suggestion
    - uses: actions/checkout@v4 
    - name: install rust toolchain
      run: rustup target add thumbv6m-none-eabi
    - name: run check
      run: cargo check -p cordyceps --target thumbv6m-none-eabi
```

---

Also, incidentally, i think the current "install rust toolchain" step is wrong _everywhere_, since it's running before checkout and doesn't include the toolchain file. it will always show the default toolchain on the github actions runner, and not the toolchain in our `rust-toolchain.toml`...so it's basically pointless. _And_, furthermore, I think the recent Rustup changes to stop implicitly installing toolchains (rust-lang/rustup#3635) would have broken this if they weren't reverted (rust-lang/rustup#4211), since `rustup show` would no longer install the required toolchain if it's not present. So, we should probably change all the "install rust toolchain" steps by:
- moving them after the checkout step, and
- making them do `rustup toolchain install` rather than `rustup show`.

But, this should be done in a separate PR.

---

Opened #528 to track